src/main/java/org/kiwiproject/dynamicproperties/annotation/DynamicField.java

```diff
@@ -53,10 +57,22 @@
      * Provide a list of possible values for this field by using the given Enum as the list. Useful if field is
      * restricted to a set of values.
      *
-     * @return the choices that should be available for this field base on the given Enum.
+     *  <b>Note:</b> Only one attribute providing choices may be specified per field.
+     *
+     * @return the choices that should be available for this field based on the given Enum.
      */
     Class<? extends Enum> choicesFromEnum() default NullEnum.class;
```

Comment:
```suggestion
    Class<? extends Enum<?> choicesFromEnum() default NullEnum.class;
```

I realize this wasn't something you changed, but adding the wildcard it eliminates the warning you get in IDEs about raw types. And, note that `Enum` itself is declared as: `class Enum<E extends Enum<E>>`